src/main/java/io/nats/client/impl/PushMessageManager.java

```diff
@@ -30,37 +30,50 @@
 
 class PushMessageManager extends MessageManager {
 
-    private static final List<Integer> PUSH_KNOWN_STATUS_CODES = Arrays.asList(409);
+    protected static final List<Integer> PUSH_KNOWN_STATUS_CODES = Collections.singletonList(409);
 
-    private static final int THRESHOLD = 3;
+    protected static final int THRESHOLD = 3;
 
-    private final NatsConnection conn;
+    protected final NatsConnection conn;
 
-    private final boolean syncMode;
-    private final boolean queueMode;
-    private final boolean hb;
-    private final boolean fc;
+    protected final NatsJetStream js;
+    protected final String stream;
+    protected final ConsumerConfiguration serverCC;
+    protected final NatsDispatcher dispatcher;
 
-    private final long idleHeartbeatSetting;
-    private final long alarmPeriodSetting;
+    protected final boolean syncMode;
+    protected final boolean queueMode;
+    protected final boolean hb;
+    protected final boolean fc;
 
-    private String lastFcSubject;
-    private long lastStreamSeq;
-    private long lastConsumerSeq;
+    protected final long idleHeartbeatSetting;
+    protected final long alarmPeriodSetting;
 
-    private final AtomicLong lastMsgReceived;
-    private HeartbeatTimer heartbeatTimer;
+    protected String lastFcSubject;
+    protected long lastStreamSeq;
+    protected long lastConsumerSeq;
 
-    PushMessageManager(NatsConnection conn, SubscribeOptions so,
-                       ConsumerConfiguration cc,
-                       boolean queueMode, boolean syncMode)
+    protected final AtomicLong lastMsgReceived;
+    protected HeartbeatTimer heartbeatTimer;
```

Comment:
more variables and protected scope for sub classing and combination of handling behaviors